packages/admin/admin-theme/src/componentsTheme/MuiDataGrid.tsx

```diff
@@ -46,21 +70,24 @@ export const getMuiDataGrid: GetMuiComponentTheme<"MuiDataGrid"> = (component, {
         root: {
             backgroundColor: "white",
         },
-        columnHeader: {
+        columnHeader: ({ ownerState }) => ({
             "&:focus": {
                 outline: "none",
             },
             "&:focus-within": {
                 outline: "none",
             },
-        },
+            ...getDensityStyles(ownerState?.density, spacing),
```

Comment:
A small detail: The design shows the height of the header to be `20px` including the `border-bottom`.
The current implementation sets the height to `20px`, then adds another pixel with the border. 

I think this is the same with the rows.